ballerina/stream_types.bal

```diff
@@ -0,0 +1,567 @@
+
+    private Endpoint[] endpoints = [];
+    private string? nextToken = ();
+    private boolean initialized = false;
+
+    public isolated function init(http:Client amazonSNSClient,
+            isolated function (map<string>) returns http:Request|Error generateRequest, 
+            string platformApplicationArn) {
+        self.amazonSNSClient = amazonSNSClient;
+        self.generateRequest = generateRequest;
+        self.platformApplicationArn = platformApplicationArn;
+    }
+
+    private isolated function fetchPlatformApplicationEndpoints() returns Error? {
+        map<string> parameters = initiateRequest("ListEndpointsByPlatformApplication");
+        parameters["PlatformApplicationArn"] = <string>self.platformApplicationArn;
+
+        if self.nextToken is string {
+            parameters["NextToken"] = <string>self.nextToken;
+        }
+
+        http:Request request = check self.generateRequest(parameters);
+        json response = check sendRequest(self.amazonSNSClient, request);
+
+        json|error nextToken = response.ListEndpointsByPlatformApplicationResponse
+            .ListEndpointsByPlatformApplicationResult.NextToken;
+
+        if nextToken is json && nextToken != () {
+            self.nextToken = nextToken.toString();
+        } else {
+            self.nextToken = ();
+        }
+
+        do {
+            json[] platformApplicationEndpoints = <json[]>(check response.ListEndpointsByPlatformApplicationResponse
+                .ListEndpointsByPlatformApplicationResult.Endpoints);
+
+            foreach json platformApplicationEndpoint in platformApplicationEndpoints {
+
+                EndpointAttributes attributes = check mapJsonToEndpointAttributes(
+                    check platformApplicationEndpoint.Attributes);
+                Endpoint endpoint = {
+                    endpointArn: check platformApplicationEndpoint.EndpointArn,
+                    ...attributes
+                };
+                self.endpoints.push(endpoint);
+
+            }
+        } on fail error e {
+            return error ResponseHandleFailedError(e.message(), e);
+        }
+    }
+
+    public isolated function next() returns record {|Endpoint value;|}|Error? {
+        if self.endpoints.length() == 0 {
+            if self.initialized && self.nextToken is () {
+                return ();
+            }
+
+            Error? e = self.fetchPlatformApplicationEndpoints();
+            self.initialized = true;
+            if e is error {
+                return e;
+            }
+        }
+
+        if self.endpoints.length() == 0 {
+            return ();
+        }
+
+        return {value: self.endpoints.remove(0)};
+    }
+
+    public isolated function close() returns Error? {
+        return ();
+    }
+}
+
+# Used to fetch and return a stream of SNS sandbox phone numbers. The logic of fetching the phone numbers is abstracted
+# away from the user.
+public class SMSSandboxPhoneNumberStream {
```

Comment:
```suggestion
class SMSSandboxPhoneNumberStream {
```

Reply:
Fixed in [d0642c9](https://github.com/ballerina-platform/module-ballerinax-aws.sns/pull/40/commits/d0642c9bacc1759708df7790f00ad058df92a123)